src/input_union_type.ts

```diff
@@ -36,31 +31,102 @@ export default (typeName: string, inputTypes: GraphQLInputObjectType[], descript
     return new GraphQLScalarType({
         name: typeName,
         description,
-        serialize: (value: any) => value,
+        serialize: (value: any) => String(value),
         parseValue: (value: any) => {
             const hasType = inputTypes.reduce((acc, t) => {
-                const result = coerceValue(value, t);
-                return result.errors && result.errors.length > 0 ? acc : true;
+                try {
+                    const result = coerceInputValue(value, t);
+                    return result.errors && result.errors.length > 0 ? acc : true;
+                } catch (error) {
+                    return acc;
+                }
             }, false);
 
             if (hasType) {
                 return value;
             }
             throw generateInputTypeError(typeName, inputTypes);
         },
+        // tslint:disable-next-line: cyclomatic-complexity
         parseLiteral: ast => {
-            const inputType = inputTypes.reduce((acc: any, type) => {
-                const astClone = JSON.parse(JSON.stringify(ast));
-                try {
-                    return isValidLiteralValue(type, astClone).length === 0 ? type : acc;
```

Comment:
`isValidLiteralValue` is deprecated in future versions of `graphql-js >= 15.0.0`. Now, we inspect the provided ObjectValue literal and determine whether it is a compound filter scalar or a filter scalar. The logic is as follows:

if object keys = in | or | not -> is a compound filter scalar
if object keys = field && operator && value -> is a filter scalar
else throw